### PR TITLE
Resolve PHP-7.4 ErrorException

### PR DESCRIPTION
**Pull request by staticglobal**
This package is crashing hard on my PHP-7.4 installation and @eukles has committed the correct fix here earlier this week with commit 45ccf52.  Maybe some of these other commits are worth looking into as well?

### CODE REVIEW

**Comment by tbjornli** [comment]
I experienced the same when upgrading to PHP 7.4 so it would be good if this pull requests could be merged in.

**Comment by bilogic** [comment]
@kulbakin can help please?

**Comment by kulbakin** [comment]
@bilogic the pull request is raw and cannot be merged as it is (namely composer.json changes, haven't checked further), i would merge if pull request wouldn't require tweaking.

File: tcpdi_parser.php [hunk 1]
```diff
@@ -1423,7 +1426,7 @@ private function _getPageRotation($obj) { // $obj = /Page
                 return false;
             } else {
                 $res = $this->_getPageRotation($obj[1][1]['/Parent']);
-                if ($res[0] == PDF_TYPE_OBJECT)
+                if (is_array($res) && $res[0] == PDF_TYPE_OBJECT)
```
Comment (bilogic) [inline tcpdi_parser.php:1429]: @staticglobal What does this do?

File: composer.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-	"name": "propa/tcpdi",
+	"name": "eukles/tcpdi",
```
Comment (bilogic) [inline composer.json:2]: @staticglobal this cannot change, it must remain as `propa/tcpdi`

**Comment by bilogic** [comment]
@kulbakin 

I went through all the changes, only ~2~ 1 item is outstanding:

1. `composer.json` name needs to be changed to your package name
2. ~Not quite sure what `if (is_array($res) && $res[0] == PDF_TYPE_OBJECT)` does~

Other than this, the rest are just better error reporting or changing of curly braces.
Hope this helps. 

Thank you.

File: tcpdi_parser.php (hunk 1 above)
Comment (bilogic) [reply tcpdi_parser.php:1429]: @kulbakin 

Actually, I think this is acceptable too, just an additional check to see if `$res` is an array.

**Comment by kulbakin** [comment]
@bilogic the second one is OK (just a safeguard to avoid error if $res is not an object), so only package name is an issue

**Comment by bilogic** [comment]
> @bilogic the second one is OK (just a safeguard to avoid error if $res is not an object), so only package name is an issue

@kulbakin correct.

**Pull request merged**